onnxruntime/core/providers/cpu/rnn/deep_cpu_gru.h

```diff
@@ -70,11 +70,7 @@ class DeepCpuGruOp final : public OpKernel {
   // across them. mutable due to this.
   // The alternative would be to create a threadpool in each call to Compute but that would incur thread creation
   // cost on every call.
-#ifdef USE_EIGEN_THREADPOOL
-  mutable Eigen::NonBlockingThreadPool ttp_{static_cast<int>(std::thread::hardware_concurrency())};
-#else
-  mutable TaskThreadPool ttp_{std::thread::hardware_concurrency()};
-#endif
+  mutable onnxruntime::concurrency::ThreadPool ttp_{"DEEPCPU_GRU", std::thread::hardware_concurrency()};
```

Comment:
>ThreadPool [](start = 36, length = 10)

Ditto How to we handle the over-subscription as many threadpools are instantiated and all do not want to go hgher than hardware concurrency?

Reply:
The core of this PR is to get each part of the system to use the same threadpool "implementation", the next PRs are to use the same threadpool, use work stealing to handle over-subscription, etc...